ctapipe_io_magic/__init__.py

```diff
@@ -888,6 +964,157 @@ def parse_metadata_info(self):
+            'MReportLaser.fIsT0ShiftFitted',
+            'MReportLaser.fIsUseGDAS',
+            'MReportLaser.fIsUpwardMoving', 
+            'MReportLaser.fOverShoot',
+            'MReportLaser.fUnderShoot',
+            'MReportLaser.fTransmission3km',
+            'MReportLaser.fTransmission6km',
+            'MReportLaser.fTransmission9km',
+            'MReportLaser.fTransmission12km',
+            'MReportLaser.fZenith',
+            'MReportLaser.fAzimuth',
+            'MReportLaser.fCloudFWHM[10]', 
+            'MReportLaser.fCloudBase[10]', 
+            'MReportLaser.fCloudTop[10]', 
+            'MReportLaser.fCloudTrans[10]', 
+            'MReportLaser.fCloudHM[10]',
+            'MReportLaser.fCloudHStd[10]',
+            'MReportLaser.fCloudLR[10]',
+            'MReportLaser.fFullOverlap',
+            'MReportLaser.fEndGroundLayer',
+            'MReportLaser.fGroundLayerTrans',
+            'MReportLaser.fKlett_k',
+            'MReportLaser.fPheCounts',
+            'MReportLaser.fCalimaness',
+            'MReportLaser.fCloudLayerAlt',
+            'MReportLaser.fCloudLayerDens',
+            'MReportLaser.fOffset',
+            'MReportLaser.fOffset_Calculated',
+            'MReportLaser.fOffset_Fitted',
+            'MReportLaser.fOffset2',
+            'MReportLaser.fOffset3',
+            'MReportLaser.fBackground1',
+            'MReportLaser.fBackground2',
+            'MReportLaser.fBackgroundErr1',
+            'MReportLaser.fBackgroundErr2',
+            'MReportLaser.fRangeMax',
+            'MReportLaser.fRangeMax_Clouds',
+            'MReportLaser.fErrorCode',
+            'MReportLaser.fScaleHeight_fit',
+            'MReportLaser.fChi2Alpha_fit',
+            'MReportLaser.fChi2Alpha_fit',
+            'MReportLaser.fAlpha_firstY',
+            'MReportLaser.fAlpha_Junge',
+            'MReportLaser.fPBLHeight',
+            'MReportLaser.fChi2Full_fit',
+            'MReportLaser.fHWSwitchMaxOffset',
+            'MReportLaser.fNCollapse',
+            'MReportLaser.fShots', 
+            'MReportLaser.fT0Shift',
+            'MReportLaser.fInterval_0',
+            'MReportLaser.fRCS_min_perfect',
+            'MReportLaser.fRCS_min_clouds',
+            'MReportLaser.fRCS_min_mol',
+            'MReportLaser.fLIDAR_ratio',
+            'MReportLaser.fLIDAR_ratio_Cloud',
+            'MReportLaser.fLIDAR_ratio_Junge',
+        ]
+
+        laser = ReportLaserContainer()  # Create an instance of ReportLaserContainer
+
+        for rootf in self.files_:
+            laser_info_runh = rootf['Laser'].arrays(
+                laser_info_array_list_runh, library="np"
+            )
+
+            # Populate the attributes of the laser object with values from report_laser
+            laser.UniqueID = laser_info_runh['MReportLaser.MReport.fUniqueID'][0]
+            laser.Bits = laser_info_runh['MReportLaser.MReport.fBits'][0]
+            laser.MJD = laser_info_runh['MTimeLaser.fMjd'][0]
```

Comment:
this needs to be modified: you should add the milisec information into the integer MJD
(and remove those fields that I wrote above earlier)